src/enclave/enclave_oe.c

```diff
@@ -61,40 +178,71 @@ void sgxlkl_ethread_init(void)
     sched_tcb->schedctx = (struct schedctx*)((char*)tls_page + tls_offset);
 
     /* Wait until libc has been initialized */
-    while (__libc_state != libc_initialized)
+    while (sgxlkl_enclave_state.libc_state != libc_initialized)
```

Comment:
`libc_state` probably needs to go away.  The kernel and libc can't share globals.  I don't think we actually need this spin at all after #403 lands.  We can start up additional lthread scheduler instances before libc starts, there just won't be any userspace threads on the queue for them to process.

Reply:
That's good to know, so I'll keep it until #403 is merged.

Reply:
In the PoC Mike and I are working on, kernel calls into user space with parameters. `sgxlkl_enclave_state` could be one of them. It seems #403 still has a long way to go, I'd recommend we merge #290 first if there is no blocking issues.

Reply:
Yeah, absolutely, #403 isn't a blocker for this PR.